codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2HeadersEncoder.java

```diff
@@ -109,7 +109,10 @@ public Configuration configuration() {
     }
 
     private void encodeHeader(ByteString key, ByteString value, OutputStream stream) throws IOException {
-        encoder.encodeHeader(stream, key.array(), value.array(), sensitivityDetector.isSensitive(key, value));
+        encoder.encodeHeader(stream,
+                key.isEntireArrayUsed() ? key.array() : new ByteString(key, true).array(),
+                value.isEntireArrayUsed() ? value.array() : new ByteString(value, true).array(),
```

Comment:
@trustin - I would like to do this, and we have an issue to track it here https://github.com/netty/netty/issues/3597.  I think others are also interested in seeing these types of changes too.